python/graphstorm/run/launch.py

```diff
@@ -349,29 +350,35 @@ def wrap_dist_remap_command(
             Number of rows per output file.
         preserve_input:
             Whether we preserve the input data.
+
+        Returns
+        -------
+        A string of remap_result launch command that combines the original arguments list
+        with remap specific arguments.
     """
     # Get the python interpreter used right now.
     # If we can not get it we go with the default `python3`
     python_bin = sys.executable \
         if sys.executable is not None and sys.executable != "" \
         else "python3 "
 
-    udf_command[0] = "graphstorm.gconstruct.remap_result"
+    new_udf_command = copy.deepcopy(udf_command)
+    new_udf_command[0] = "graphstorm.gconstruct.remap_result"
 
     # Add remap related arguments
-    udf_command += ["--rank", str(rank)]
-    udf_command += ["--world-size", str(world_size)]
-    udf_command += ["--with-shared-fs", "True" if with_shared_fs else "False"]
-    udf_command += ["--num-processes", str(num_trainers)]
-    udf_command += ["--output-chunk-size", str(output_chunk_size)]
-    udf_command += ["--preserve-input", "True" if preserve_input else "False"]
+    new_udf_command += ["--rank", str(rank)]
+    new_udf_command += ["--world-size", str(world_size)]
+    new_udf_command += ["--with-shared-fs", "True" if with_shared_fs else "False"]
+    new_udf_command += ["--num-processes", str(num_trainers)]
+    new_udf_command += ["--output-chunk-size", str(output_chunk_size)]
+    new_udf_command += ["--preserve-input", "True" if preserve_input else "False"]
```

Comment:
Actually maybe we just write:

```suggestion
    new_udf_command.extend([
        "--rank", str(rank),
        "--world-size", str(world_size),
        "--with-shared-fs", "True" if with_shared_fs else "False",
        "--num-processes", str(os.cpu_count()),
        "--output-chunk-size", str(output_chunk_size),
        "--preserve-input", "True" if preserve_input else "False"]
    )
```

Seems more readable to me at least. Not that we also change the default num-processes

Reply:
I would suggest we create another PR for setting num-processes to os.cpu_count(). And we need to do some profiling to see whether os.cpu_count() is the right default number.